### PR TITLE
JAVA-2976 Protocol v5 error codes CAS_WRITE_UNKNOWN, CDC_WRITE_FAILURE not supported (4.x)

### PR DESCRIPTION
**Pull request by absurdfarce**
4.x equivalent to https://github.com/datastax/java-driver/pull/1585.  Relies on https://github.com/datastax/native-protocol/pull/42 to build and function correctly.

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
       <dependency>
         <groupId>com.datastax.oss</groupId>
         <artifactId>native-protocol</artifactId>
-        <version>1.5.0</version>
+        <version>1.5.1-SNAPSHOT</version>
```
Comment (absurdfarce) [inline bom/pom.xml]: Just a placeholder for now to use a locally-built version of native-protocol that includes https://github.com/datastax/native-protocol/pull/42.  To be replaced by an actual version once we release a new version of native-protocol.

File: core/src/main/java/com/datastax/oss/driver/api/core/servererrors/CASWriteUnknownException.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import com.datastax.oss.driver.api.core.cql.ExecutionInfo;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.retry.RetryPolicy;
+import com.datastax.oss.driver.api.core.session.Request;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * The result of a CAS operation is in an unknown state
+ *
+ * <p>This exception is processed by {@link RetryPolicy#onReadTimeoutVerdict(Request,
```
Comment (adutra) [inline core/src/main/java/com/datastax/oss/driver/api/core/servererrors/CASWriteUnknownException.java]: This is not accurate, it is processed by `onErrorResponseVerdict`.

**Review by adutra — CHANGES_REQUESTED** [review]
I thought it would be easier to add tests in 4.x than in 3.x, but looking at existing tests, we seem to have a grand total of zero unit tests for specific error codes. I guess we can live we that since the codes were tested in native-protocol. It's just the translation from error code to exception that is not covered, but that's minor.

**Comment by adutra** [comment]
FYI I ran a build on our internal CI off of this branch, it was clean.

File: core/src/main/java/com/datastax/oss/driver/api/core/servererrors/CASWriteUnknownException.java (hunk 2 above)
Comment (absurdfarce) [reply core/src/main/java/com/datastax/oss/driver/api/core/servererrors/CASWriteUnknownException.java]: Ah, another good catch... and another copy/paste error.  Fixing.

File: core/src/main/java/com/datastax/oss/driver/api/core/servererrors/CASWriteUnknownException.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.DriverException;
+import com.datastax.oss.driver.api.core.cql.ExecutionInfo;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.retry.RetryPolicy;
+import com.datastax.oss.driver.api.core.session.Request;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * The result of a CAS operation is in an unknown state
```
Comment (adutra) [inline core/src/main/java/com/datastax/oss/driver/api/core/servererrors/CASWriteUnknownException.java]:
```suggestion
 * The result of a CAS operation is in an unknown state.
```

**Review by adutra — APPROVED** [review]

File: bom/pom.xml (hunk 1 above)
Comment (absurdfarce) [reply bom/pom.xml]: native-protocol 1.5.1 has now been released

**Pull request merged**